cmd/snap/cmd_list.go

```diff
@@ -64,6 +65,42 @@ func (x *cmdList) Execute(args []string) error {
 
 var ErrNoMatchingSnaps = errors.New(i18n.G("no matching snaps installed"))
 
+// snapd will give us  and we want
+// "" (local snap)     "-"
+// risk                risk
+// track               track        (not yet returned by snapd)
+// track/stable        track
+// track/risk          track/risk
+// risk/branch         risk/…
+// track/risk/branch   track/risk/…
+func fmtChannel(ch string) string {
+	if ch == "" {
+		// local snap
+		return "-"
+	}
+	idx := strings.IndexByte(ch, '/')
+	if idx < 0 {
+		// risk
+		return ch
+	}
+	first, rest := ch[:idx], ch[idx+1:]
+	if rest == "stable" && first != "" {
+		// first is a track
+		return first
+	}
+	if idx2 := strings.IndexByte(rest, '/'); idx2 >= 0 {
```

Comment:
I personally find I easier to read if you the place of this `if..` with the `for..` loop that directly follows it.